From 9eef094cfc9e7446527b283de3606bd8929aab80 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 6 Sep 2022 07:43:31 +0800
Subject: [PATCH 45/60] Revert "drm/bridge: tc358767: Move (e)DP bridge
 endpoint parsing into dedicated function"

This reverts commit c038b9b7338924e5eb8d4b34d1a76691529feef8.
---
 drivers/gpu/drm/bridge/tc358767.c | 30 +++++++++---------------------
 1 file changed, 9 insertions(+), 21 deletions(-)

diff --git a/drivers/gpu/drm/bridge/tc358767.c b/drivers/gpu/drm/bridge/tc358767.c
index 9af564cf92e50..34a3e4e9f7175 100644
--- a/drivers/gpu/drm/bridge/tc358767.c
+++ b/drivers/gpu/drm/bridge/tc358767.c
@@ -1535,12 +1535,19 @@ static irqreturn_t tc_irq_handler(int irq, void *arg)
 	return IRQ_HANDLED;
 }
 
-static int tc_probe_edp_bridge_endpoint(struct tc_data *tc)
+static int tc_probe(struct i2c_client *client, const struct i2c_device_id *id)
 {
-	struct device *dev = tc->dev;
+	struct device *dev = &client->dev;
 	struct drm_panel *panel;
+	struct tc_data *tc;
 	int ret;
 
+	tc = devm_kzalloc(dev, sizeof(*tc), GFP_KERNEL);
+	if (!tc)
+		return -ENOMEM;
+
+	tc->dev = dev;
+
 	/* port@2 is the output port */
 	ret = drm_of_find_panel_or_bridge(dev->of_node, 2, 0, &panel, NULL);
 	if (ret && ret != -ENODEV)
@@ -1559,25 +1566,6 @@ static int tc_probe_edp_bridge_endpoint(struct tc_data *tc)
 		tc->bridge.type = DRM_MODE_CONNECTOR_DisplayPort;
 	}
 
-	return ret;
-}
-
-static int tc_probe(struct i2c_client *client, const struct i2c_device_id *id)
-{
-	struct device *dev = &client->dev;
-	struct tc_data *tc;
-	int ret;
-
-	tc = devm_kzalloc(dev, sizeof(*tc), GFP_KERNEL);
-	if (!tc)
-		return -ENOMEM;
-
-	tc->dev = dev;
-
-	ret = tc_probe_edp_bridge_endpoint(tc);
-	if (ret)
-		return ret;
-
 	/* Shut down GPIO is optional */
 	tc->sd_gpio = devm_gpiod_get_optional(dev, "shutdown", GPIOD_OUT_HIGH);
 	if (IS_ERR(tc->sd_gpio))
-- 
2.25.1

